xapian-core/tests/api_weight.cc

```diff
@@ -134,6 +134,66 @@ DEFINE_TESTCASE(bm25weight5, backend) {
 }
 
 // Test exception for junk after serialised weight.
+DEFINE_TESTCASE(bm25plusweight1, !backend) {
+    Xapian::BM25PlusWeight wt(2.0, 0.5, 1.3, 0.6, 0.01, 0.5);
+    try {
+    Xapian::BM25PlusWeight b;
+    Xapian::BM25PlusWeight * b2 = b.unserialise(wt.serialise() + "X");
+    // Make sure we actually use the weight.
+    bool empty = b2->name().empty();
+    delete b2;
+    if (empty)
+	FAIL_TEST("Serialised BM25PlusWeight with junk appended unserialised to empty name!");
+    FAIL_TEST("Serialised BM25PlusWeight with junk appended unserialised OK");
+    } catch (const Xapian::SerialisationError &) {
+    // Good!
+    }
```

Comment:
The bodies of both the `try` and `catch` blocks are under-indented here (unlike the code of the `bm25weight3` testcase which this seems to be based on).